### PR TITLE
Add export and inference glm

### PR DESCRIPTION
**Pull request by wawltor**
### PR types
New features 
### PR changes
Models

### Description
Add export glm

### CODE REVIEW

**Review by gongel — DISMISSED** [review]

File: paddlenlp/transformers/glm/modeling.py [hunk 1]
```diff
@@ -793,28 +793,29 @@ def prepare_inputs_for_generation(
         self,
         input_ids: Tensor,
         position_ids: Tensor = None,
-        generation_attention_mask: Tensor = None,
+        attention_mask: Tensor = None,
         cache: Tensor = None,
         **kwargs
     ):
-        attention_mask = generation_attention_mask
+        attention_mask_gen = attention_mask
```
Comment (sijunhe) [inline paddlenlp/transformers/glm/modeling.py:800]: 这里直接用attention_mask吧，attention_mask_gen就不用了

File: paddlenlp/transformers/glm/tokenizer.py [hunk 2]
```diff
@@ -209,12 +209,12 @@ def build_inputs_for_generation(
...
         batch = {"input_ids": input_ids, "position_ids": position_ids}
         if labels is None:
-            batch["generation_attention_mask"] = attention_mask
+            batch["attention_mask"] = attention_mask
         else:
             batch["attention_mask"] = attention_mask
             batch["loss_mask"] = loss_mask
             batch["label_ids"] = labels
-        return BatchEncoding(batch, tensor_type="pd")
+        return BatchEncoding(batch, tensor_type="np")
```
Comment (sijunhe) [inline paddlenlp/transformers/glm/tokenizer.py:217]: 这里我有一个需求，看看你这个PR能不能一并满足：
目前你的实现是prompt pad到统一长度input length，labels pad到统一长度gen_length。但是其实labels可以不用pad到同一长度，只需要做一个max length的截断，然后pad的工作可以交给collator

File: paddlenlp/transformers/glm/modeling.py (hunk 1 above)
Comment (wawltor) [reply paddlenlp/transformers/glm/modeling.py:800]: > 这里直接用attention_mask吧，attention_mask_gen就不用了

不可以直接用attention_mask，这里的attention_mask是全局的，每次预测的时候需要在attention_mask进行切割

File: paddlenlp/transformers/glm/tokenizer.py (hunk 2 above)
Comment (wawltor) [reply paddlenlp/transformers/glm/tokenizer.py:217]: 这里的tokenizer是可以支持不设置Padding，可以在Collator部分进行适配

**Review by sijunhe — APPROVED** [review]
lgtm
